clients/web/test/testUtils.js

```diff
@@ -11,6 +11,17 @@ window.alert = function (msg) {
     expect('Alert was used').toBe('Alerts should not be present');
 };
 
+// Augment jQuery's click() function to follow href properties in links if they exist.
+$.fn.click = _.wrap($.fn.click, function (old) {
+    var args = Array.prototype.slice.call(arguments, 1);
```

Comment:
Why can't you do `arguments.slice(1)` here?

Reply:
The silliness of javascript: `arguments` is not an `Array`.

Reply:
Ah, I see now.

However, it appears that this usage of `arguments` is a known optimization killer:
* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments
* https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#3-managing-arguments
I think it's fine in testing, but should definitely never make it to production. Could you add a code comment stating so?

Reply:
This whole patch might be going away, @ronichoudhury and I were running into spooky behavior with the last failing test and we may need a different workaround entirely. It's premature to review any of the commits on this branch that are fixing up the tests.

Reply:
Got it, no problem.